### PR TITLE
Remove bad chain alert partition check

### PR DESCRIPTION
**Pull request by LarryRuane**
(cherry picked from https://github.com/bitcoin/bitcoin/pull/8275 commit ab8be98fdb25b678a8cd7e89adf06d1b1f6bdd62), original commit message:

As per meeting 2016-03-31
https://bitcoincore.org/en/meetings/2016/03/31/#bad-chain-alerts

The partition checker was producing huge number of false-positives
and was disabled in 0.12.1 on the understanding it would either be
fixed in 0.13 or removed entirely from master if not.

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
<strike>utACK</strike>

File: qa/rpc-tests/hardforkdetection.py [hunk 1]
```diff
@@ -47,10 +42,6 @@ def run_test(self):
         # Check that safe mode is on.
         self.assert_safemode_on("We do not appear to fully agree with our peers!")
 
-        # Check that an -alertnotify was triggered.
-        with open(self.alert_filename, 'r', encoding='utf8') as f:
-            alert_text = f.read()
-
         if len(alert_text) == 0:
             raise AssertionError("-alertnotify did not warn of detected hard fork")
```
Comment (daira) [inline qa/rpc-tests/hardforkdetection.py:47]:
```suggestion
```

File: qa/rpc-tests/hardforkdetection.py [hunk 2]
```diff
@@ -12,15 +12,10 @@
...
         self.nodes = []
-        self.alert_filename = os.path.join(self.options.tmpdir, "alert.txt")
-        with open(self.alert_filename, 'w', encoding='utf8'):
-            pass  # Just open then close to create zero-length file
         self.nodes.append(start_node(0, self.options.tmpdir,
-                            ["-blockversion=2", "-alertnotify=echo %s >> \"" + self.alert_filename + "\""]))
+                            ["-blockversion=2"]))
 
     def assert_safemode_off(self):
         self.nodes[0].getbalance()
```
Comment (nuttycom) [inline qa/rpc-tests/hardforkdetection.py:21]:
```suggestion
        # if safemode is on, getbalance() will throw
        self.nodes[0].getbalance()
```

File: qa/rpc-tests/hardforkdetection.py [hunk 3]
```diff
@@ -47,10 +42,6 @@ def run_test(self):
         # Check that safe mode is on.
         self.assert_safemode_on("We do not appear to fully agree with our peers!")
 
-        # Check that an -alertnotify was triggered.
-        with open(self.alert_filename, 'r', encoding='utf8') as f:
-            alert_text = f.read()
-
```
Comment (str4d) [inline qa/rpc-tests/hardforkdetection.py:53]: We still have the logic that generates this, in `CheckForkWarningConditions`. Undo the changes to this test.

**Comment by LarryRuane** [comment]
Addressed review comments, and made this cherry-pick as close as possible to the upstream PR.

**Comment by nuttycom** [comment]
The hardforkdetection.py test is failing for me:

```
Running individually selected tests: 
	hardforkdetection.py
...................................................................
hardforkdetection.py:
Initializing test directory /tmp/testdy24w8p4/186293
initialize_chain(): Removing stale cache
JSONRPC error: CreateNewBlock(): TestBlockValidity failed: bad-version
Stopping nodes
Cleaning up
Failed

stderr:
  File "/home/nuttycom/work/zcash/work/qa/rpc-tests/test_framework/test_framework.py", line 143, in main
    self.run_test()
  File "/home/nuttycom/work/zcash/work/qa/rpc-tests/hardforkdetection.py", line 40, in run_test
    self.nodes[0].generate(100)
  File "/home/nuttycom/work/zcash/work/qa/rpc-tests/test_framework/coverage.py", line 49, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/home/nuttycom/work/zcash/work/qa/rpc-tests/test_framework/authproxy.py", line 137, in __call__
    raise JSONRPCException(response['error'])

Pass: False, Duration: 34 s

TEST                 | PASSED | DURATION

hardforkdetection.py | False  | 34 s

ALL                  | False  | 34 s (accumulated)

Runtime: 34 s
```

**Comment by LarryRuane** [comment]
> The hardforkdetection.py test is failing for me

Yes, this is one of the tests that has been broken for a long time (fails the same way without the current PR), it's part of the `EXTENDED_SCRIPTS` list (in `rpc-tests.py`) that aren't run by default (and not run by CI). Maybe we can fix this and other tests in a separate PR.

File: qa/rpc-tests/hardforkdetection.py (hunk 2 above)
Comment (LarryRuane) [reply qa/rpc-tests/hardforkdetection.py]: This is a good suggestion (I didn't understand that myself), but since this PR is no longer touching this file, I think it's best to leave this change out.

**Review by str4d — APPROVED** [review]
utACK

**Comment by str4d** [comment]
Waiting to r+ this until CI is fixed.

**Comment by str4d** [comment]
@zkbot r+
